qa/rolling-upgrade/build.gradle

```diff
@@ -29,112 +30,102 @@ task bwcTest {
+    nonInputProperties.systemProperty('tests.clustername', "${-> testClusters."${baseName}".getName() }")
   }
 
-  Task twoThirdsUpgradedTest = tasks.create(name: "${baseName}#twoThirdsUpgradedTest", type: RestIntegTestTask)
-
-  configureUpgradeCluster("twoThirdsUpgradedTestCluster", oneThirdUpgradedTestRunner, 1,
-    // Use all running nodes as seed nodes so there is no race between pinging and the tests
-    { [oldClusterTest.nodes.get(2).transportUri(), oneThirdUpgradedTest.nodes.get(0).transportUri()] })
-
-  Task twoThirdsUpgradedTestRunner = tasks.getByName("${baseName}#twoThirdsUpgradedTestRunner")
-  twoThirdsUpgradedTestRunner.configure {
+  tasks.register("${baseName}#twoThirdsUpgradedTest", RestTestRunnerTask) {
+    dependsOn "${baseName}#oneThirdUpgradedTest"
+    useCluster testClusters."${baseName}"
+    doFirst {
+      testClusters."${baseName}".nextNodeToNextVersion()
+    }
     systemProperty 'tests.rest.suite', 'mixed_cluster'
     systemProperty 'tests.first_round', 'false'
-    finalizedBy "${baseName}#oldClusterTestCluster#node2.stop"
+    nonInputProperties.systemProperty('tests.rest.cluster', "${-> testClusters."${baseName}".allHttpSocketURI.join(",") }")
+    nonInputProperties.systemProperty('tests.clustername', "${-> testClusters."${baseName}".getName() }")
   }
 
-  Task upgradedClusterTest = tasks.create(name: "${baseName}#upgradedClusterTest", type: RestIntegTestTask)
-
-  configureUpgradeCluster("upgradedClusterTestCluster", twoThirdsUpgradedTestRunner, 2,
-    // Use all running nodes as seed nodes so there is no race between pinging and the tests
-    { [oneThirdUpgradedTest.nodes.get(0).transportUri(), twoThirdsUpgradedTest.nodes.get(0).transportUri()] })
-
-  Task upgradedClusterTestRunner = tasks.getByName("${baseName}#upgradedClusterTestRunner")
-  upgradedClusterTestRunner.configure {
+  tasks.register("${baseName}#upgradedClusterTest", RestTestRunnerTask) {
+    dependsOn "${baseName}#twoThirdsUpgradedTest"
+    doFirst {
+      testClusters."${baseName}".nextNodeToNextVersion()
+    }
+    useCluster testClusters."${baseName}"
     systemProperty 'tests.rest.suite', 'upgraded_cluster'
-    /*
-     * Force stopping all the upgraded nodes after the test runner
-     * so they are alive during the test.
-     */
-    finalizedBy "${baseName}#oneThirdUpgradedTestCluster#stop"
-    finalizedBy "${baseName}#twoThirdsUpgradedTestCluster#stop"
-  }
-
-  Task versionBwcTest = tasks.create(name: "${baseName}#bwcTest") {
-    enabled = project.bwc_tests_enabled
-    dependsOn = [upgradedClusterTest]
+    nonInputProperties.systemProperty('tests.rest.cluster', "${-> testClusters."${baseName}".allHttpSocketURI.join(",") }")
+    nonInputProperties.systemProperty('tests.clustername', "${-> testClusters."${baseName}".getName() }")
   }
 
   if (project.bwc_tests_enabled) {
-    bwcTest.dependsOn(versionBwcTest)
+    bwcTest.dependsOn(
+            tasks.register("${baseName}#bwcTest") {
```

Comment:
nit: can we fix this weird indentation here?

Reply:
I agree it's weird but it's what IDEA suggests, I prefer to keep it so it doesn't go back and forth as we edit the file.